### PR TITLE
Add Path.safe_relative/1 and Path.safe_relative_to/2

### PR DESCRIPTION
**Pull request by whatyouhide**
Part of https://github.com/elixir-lang/elixir/issues/9603.

### CODE REVIEW

File: lib/elixir/lib/path.ex [hunk 1]
```diff
@@ -721,4 +721,75 @@ defmodule Path do
...
+
+      iex> Path.safe_relative_to("my_dep/../..", "deps")
+      :error
+
+      iex> Path.safe_relative_to("/usr/local", ".")
+      :error
+
+  """
+  @doc since: "1.14.0"
+  @spec safe_relative_to(t, t) :: {:ok, binary} | {:error, :unsafe}
```
Comment (fertapric) [inline lib/elixir/lib/path.ex:757]:
```suggestion
  @spec safe_relative_to(t, t) :: {:ok, binary} | :error
```

File: lib/elixir/lib/path.ex [hunk 2]
```diff
@@ -721,4 +721,75 @@ defmodule Path do
...
+
+      iex> Path.safe_relative("foo/../..")
+      :error
+
+      iex> Path.safe_relative("/usr/local")
+      :error
+
+  """
+  @doc since: "1.14.0"
+  @spec safe_relative(t) :: {:ok, binary} | {:error, :unsafe}
```
Comment (fertapric) [inline lib/elixir/lib/path.ex:791]:
```suggestion
  @spec safe_relative(t) :: {:ok, binary} | :error
```

File: lib/elixir/lib/path.ex [hunk 3]
```diff
@@ -721,4 +721,75 @@ defmodule Path do
   defp major_os_type do
     :os.type() |> elem(0)
   end
+
+  @doc """
+  Returns a relative path that is protected from directory-traversal attacks.
+  Returns `path` relative to `relative_to`, but protecting from directory traversal attacks.
```
Comment (josevalim) [inline lib/elixir/lib/path.ex:727]: Duplicated sentences?

Comment (whatyouhide) [reply lib/elixir/lib/path.ex:727]: Indeed! Fixed.

**Review by josevalim — APPROVED** [review]

**Comment by michallepicki** [comment]
Any idea what's up with the test failure on Windows?

**Pull request merged**